apps/meteor/app/livechat/server/lib/contacts/resolveContactConflicts.ts

```diff
@@ -0,0 +1,66 @@
+import type { ILivechatContact, ILivechatContactConflictingField } from '@rocket.chat/core-typings';
+import { LivechatContacts, Settings } from '@rocket.chat/models';
+
+import { validateContactManager } from './validateContactManager';
+import { notifyOnSettingChanged } from '../../../../lib/server/lib/notifyListener';
+
+export type ResolveContactConflictsParams = {
+	contactId: string;
+	name?: string;
+	customFields?: Record<string, unknown>;
+	contactManager?: string;
+	wipeConflicts?: boolean;
+};
+
+export async function resolveContactConflicts(params: ResolveContactConflictsParams): Promise<ILivechatContact> {
+	const { contactId, name, customFields, contactManager, wipeConflicts } = params;
+
+	const contact = await LivechatContacts.findOneById<Pick<ILivechatContact, '_id' | 'customFields' | 'conflictingFields'>>(contactId, {
+		projection: { _id: 1, customFields: 1, conflictingFields: 1 },
+	});
+
+	if (!contact) {
+		throw new Error('error-contact-not-found');
+	}
+
+	if (!contact.conflictingFields) {
```

Comment:
```suggestion
	if (!contact.conflictingFields?.length) {
```

if conflictingFields ends up being an empty array, this will not stop here